types.go

```diff
@@ -108,20 +108,16 @@ type FunctionInfo struct {
 
 // functionInfo contains information about which interfaces the objective
 // function F implements and the actual methods of F that have been
-// successfully type switched.
+// successfully type switched. TODO(jds) change this comment
 type functionInfo struct {
```

Comment:
I think I had two imagined use cases. The first was that the method could test that it could handle the kind of optimization problem. We implemented that with the Needs method instead.

Another possible use would be in Global, having a "random restart Local method". If it wanted to choose a default local method, it would need to know what functions were non-nil.

It seems like there would be optimizers that would change behavior dependent on what kind of problem it is.